### ISSUE TITLE
Diagram: Editable connections size 0 after events

### ISSUE THREAD

**Issue by ksreyr** [primefaces/primefaces#5811]
## 1) Environment
- PrimeFaces version: 8.0 maven Repository
- Does it work on the newest released PrimeFaces version? Version?:  Not
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source): Not
- Application server + version: Tested in (Tomcat 8. in work) and (Wildfly 17.0.0 just to pruebe in a control Environment)

- Affected browsers: Chrome Safari and Firefox (Tested)

## 2) Expected behavior
the size in the variable connections in (this->model->connections) should be 1 

## 3) Actual behavior
after the connection process is the size in the variable connections 0 even when there is a connection Visible through the Browser.

## 4) Steps to reproduce
1: make a simple connection between two ports
2: change this connection to other port.
3: disconnect it.
:::now we are in the bug mode
4: when you try to make again the simple connection, the size of connections is 0 that is a problem because even when the confirmation message is displayed "Connected", there is not information in the variable to save and this Connection is fake, inasmuch as the connection size is 0 and the Connections in visible in Display.

I have tested that for my work and too I have made a a simple example in other application server in order to see if it works but it does not.

I have make a video in order to explain better the situation. 
[PrimefacesBug.mp4.zip](https://github.com/primefaces/primefaces/files/4496424/PrimefacesBug.mp4.zip)

I am available too to give more information about this bug. it is necessary. 

Thanks for your time.

## 5) Sample XHTML
IMPORTANT THIS IS THE SAME EXAMPLE IN THE SHOW CASES OF PRIMEFACE, I HAVE TESTED THE BUG WITH THE EXAMPLE.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://xmlns.jcp.org/jsf/html"
      xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
      xmlns:f="http://xmlns.jcp.org/jsf/core"
      xmlns:p="http://primefaces.org/ui">
<h:head>
    <title>prueba</title>
</h:head>
<h:body>
    <style type="text/css">
        .ui-diagram-element {
            border: 0.1em dotted #E5E4E2;
            background-color: #EFEEEC;
            width: 10em;
            height: 8em;
            text-align: center;
            box-shadow: 0 5px 10px rgba(0, 0, 0, 0.8);
        }

        .ui-diagram-element:hover {
            background-color: #C7C6C4;
        }
    </style>

    <h:form id="form">
        <p:growl id="msgs" showDetail="true" skipDetailIfEqualsSummary="true"/>

        <p:diagram value="#{diagramEditableView.model}" style="height:600px" styleClass="ui-widget-content" var="el">
            <f:facet name="element">
                <h:outputText value="#{el.name}" style="display:block;margin-top:1em;"/>
                <p:graphicImage name="demo/images/#{el.image}"/>
            </f:facet>
            <p:ajax event="connect" listener="#{diagramEditableView.onConnect}"/>
            <p:ajax event="disconnect" listener="#{diagramEditableView.onDisconnect}"/>
            <p:ajax event="connectionChange" listener="#{diagramEditableView.onConnectionChange}"/>
        </p:diagram>
    </h:form>
</h:body>
</html>
```

## 6) Sample bean
IMPORTANT THIS IS THE SAME EXAMPLE IN THE SHOW CASES OF PRIMEFACE, I HAVE TESTED THE BUG WITH THE EXAMPLE.
package beans;

```java
import org.primefaces.PrimeFaces;
import org.primefaces.event.diagram.ConnectEvent;
import org.primefaces.event.diagram.ConnectionChangeEvent;
import org.primefaces.event.diagram.DisconnectEvent;
import org.primefaces.model.diagram.DefaultDiagramModel;
import org.primefaces.model.diagram.DiagramModel;
import org.primefaces.model.diagram.Element;
import org.primefaces.model.diagram.connector.StraightConnector;
import org.primefaces.model.diagram.endpoint.DotEndPoint;
import org.primefaces.model.diagram.endpoint.EndPoint;
import org.primefaces.model.diagram.endpoint.EndPointAnchor;
import org.primefaces.model.diagram.endpoint.RectangleEndPoint;
import org.primefaces.model.diagram.overlay.ArrowOverlay;

import javax.annotation.PostConstruct;
import javax.faces.application.FacesMessage;
import javax.faces.context.FacesContext;
import javax.faces.view.ViewScoped;
import javax.inject.Named;
import java.io.Serializable;

@Named("diagramEditableView")
@ViewScoped
public class Editable implements Serializable {

    private DefaultDiagramModel model;

    private boolean suspendEvent;

    @PostConstruct
    public void init() {
        model = new DefaultDiagramModel();
        model.setMaxConnections(-1);

        model.getDefaultConnectionOverlays().add(new ArrowOverlay(20, 20, 1, 1));
        StraightConnector connector = new StraightConnector();
        connector.setPaintStyle("{strokeStyle:'#98AFC7', lineWidth:3}");
        connector.setHoverPaintStyle("{strokeStyle:'#5C738B'}");
        model.setDefaultConnector(connector);

        Element computerA = new Element(new NetworkElement("Computer A", "computer-icon.png"), "10em", "6em");
        EndPoint endPointCA = createRectangleEndPoint(EndPointAnchor.BOTTOM);
        endPointCA.setSource(true);
        computerA.addEndPoint(endPointCA);

        Element computerB = new Element(new NetworkElement("Computer B", "computer-icon.png"), "25em", "6em");
        EndPoint endPointCB = createRectangleEndPoint(EndPointAnchor.BOTTOM);
        endPointCB.setSource(true);
        computerB.addEndPoint(endPointCB);

        Element computerC = new Element(new NetworkElement("Computer C", "computer-icon.png"), "40em", "6em");
        EndPoint endPointCC = createRectangleEndPoint(EndPointAnchor.BOTTOM);
        endPointCC.setSource(true);
        computerC.addEndPoint(endPointCC);

        Element serverA = new Element(new NetworkElement("Server A", "server-icon.png"), "15em", "24em");
        EndPoint endPointSA = createDotEndPoint(EndPointAnchor.AUTO_DEFAULT);
        serverA.setDraggable(false);
        endPointSA.setTarget(true);
        serverA.addEndPoint(endPointSA);

        Element serverB = new Element(new NetworkElement("Server B", "server-icon.png"), "35em", "24em");
        EndPoint endPointSB = createDotEndPoint(EndPointAnchor.AUTO_DEFAULT);
        serverB.setDraggable(false);
        endPointSB.setTarget(true);
        serverB.addEndPoint(endPointSB);

        model.addElement(computerA);
        model.addElement(computerB);
        model.addElement(computerC);
        model.addElement(serverA);
        model.addElement(serverB);
    }

    public DiagramModel getModel() {
        return model;
    }

    public void onConnect(ConnectEvent event) {
        if(!suspendEvent) {
            FacesMessage msg = new FacesMessage(FacesMessage.SEVERITY_INFO, "Connected",
                    "From " + event.getSourceElement().getData()+ " To " + event.getTargetElement().getData());

            FacesContext.getCurrentInstance().addMessage(null, msg);

            PrimeFaces.current().ajax().update("form:msgs");
        }
        else {
            suspendEvent = false;
        }
    }

    public void onDisconnect(DisconnectEvent event) {
        FacesMessage msg = new FacesMessage(FacesMessage.SEVERITY_INFO, "Disconnected",
                "From " + event.getSourceElement().getData()+ " To " + event.getTargetElement().getData());

        FacesContext.getCurrentInstance().addMessage(null, msg);

        PrimeFaces.current().ajax().update("form:msgs");
    }

    public void onConnectionChange(ConnectionChangeEvent event) {
        FacesMessage msg = new FacesMessage(FacesMessage.SEVERITY_INFO, "Connection Changed",
                "Original Source:" + event.getOriginalSourceElement().getData() +
                        ", New Source: " + event.getNewSourceElement().getData() +
                        ", Original Target: " + event.getOriginalTargetElement().getData() +
                        ", New Target: " + event.getNewTargetElement().getData());

        FacesContext.getCurrentInstance().addMessage(null, msg);

        PrimeFaces.current().ajax().update("form:msgs");
        suspendEvent = true;
    }

    private EndPoint createDotEndPoint(EndPointAnchor anchor) {
        DotEndPoint endPoint = new DotEndPoint(anchor);
        endPoint.setScope("network");
        endPoint.setTarget(true);
        endPoint.setStyle("{fillStyle:'#98AFC7'}");
        endPoint.setHoverStyle("{fillStyle:'#5C738B'}");

        return endPoint;
    }

    private EndPoint createRectangleEndPoint(EndPointAnchor anchor) {
        RectangleEndPoint endPoint = new RectangleEndPoint(anchor);
        endPoint.setScope("network");
        endPoint.setSource(true);
        endPoint.setStyle("{fillStyle:'#98AFC7'}");
        endPoint.setHoverStyle("{fillStyle:'#5C738B'}");

        return endPoint;
    }

    public class NetworkElement implements Serializable {

        private String name;
        private String image;

        public NetworkElement() {
        }

        public NetworkElement(String name, String image) {
            this.name = name;
            this.image = image;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getImage() {
            return image;
        }

        public void setImage(String image) {
            this.image = image;
        }

        @Override
        public String toString() {
            return name;
        }

    }
}
```

**Comment by melloware** [comment]
@SantiRey have you debugged the issue in the PF code enough to provide a fix?

**Comment by ksreyr** [comment]
> @SantiRey have you debugged the issue in the PF code enough to provide a fix?

unfortunately not, since at work I only had to argue because the Bug existed, I realized that the error is in the JavaScript component, which I have not had access to.

If I can help in any way, please tell me and I would gladly help repair this Bug in my free time.

**Comment by melloware** [comment]
Just add this JS to your application to fix it right now.  It will be fixed for 9.0.
```javascript
PrimeFaces.widget.Diagram.prototype.onConnect = function(info) {
    var options = {
        source: this.id,
        process: this.id,
        params: [{
                name: this.id + '_connect',
                value: true
            },
            {
                name: this.id + '_sourceId',
                value: info.sourceId.substring(this.id.length + 1)
            },
            {
                name: this.id + '_targetId',
                value: info.targetId.substring(this.id.length + 1)
            },
            {
                name: this.id + '_sourceEndPointId',
                value: info.sourceEndpoint.getUuid()
            },
            {
                name: this.id + '_targetEndPointId',
                value: info.targetEndpoint.getUuid()
            }
        ]
    };

    if (this.connectionChanged) {
        options.params.push({
            name: this.id + '_connectionChanged',
            value: true
        });
    }
    this.connectionChanged = false;

    if (this.hasBehavior('connect')) {
        this.callBehavior('connect', options);
    } else {
        PrimeFaces.ajax.Request.handle(options);
    }
};
```
